include/nbl/builtin/hlsl/bxdf/ndf/beckmann.hlsl

```diff
@@ -16,46 +16,47 @@ namespace bxdf
 namespace ndf
 {
 
-// template<typename T, bool IsAniso NBL_STRUCT_CONSTRAINABLE>
-// struct Beckmann;
+// forward declare
+// template<typename NDF, MicrofacetTransformTypes reflect_refract>
+// struct microfacet_to_light_measure_transform;
 
-// TODO: get beta from lgamma, see: https://www.cec.uchile.cl/cinetica/pcordero/MC_libros/NumericalRecipesinC.pdf
+template<typename T, bool IsAnisotropic=false NBL_STRUCT_CONSTRAINABLE>
+struct Beckmann;
 
 // TODO: use query_type for D, lambda, beta, DG1 when that's implemented
 template<typename T>
 NBL_PARTIAL_REQ_TOP(concepts::FloatingPointScalar<T>)
 struct Beckmann<T,false NBL_PARTIAL_REQ_BOT(concepts::FloatingPointScalar<T>) >
 {
     using scalar_type = T;
-    using this_t = Beckmann<T,false>;
 
-    scalar_type D(scalar_type a2, scalar_type NdotH2)
+    scalar_type D(scalar_type NdotH2)
     {
-        scalar_type nom = exp<scalar_type>( (NdotH2 - scalar_type(1.0)) / (a2 * NdotH2) );   // exp(x) == exp2(x/log(2)) ?
+        scalar_type nom = exp2<scalar_type>((NdotH2 - scalar_type(1.0)) / (log<scalar_type>(2.0) * a2 * NdotH2));
         scalar_type denom = a2 * NdotH2 * NdotH2;
         return numbers::inv_pi<scalar_type> * nom / denom;
```

Comment:
https://github.com/Devsh-Graphics-Programming/Nabla/pull/899#discussion_r2195206734

Reply:
actually not a concern, exp2 should handle everything INF related just fine